pkg/kubelet/lifecycle/predicate.go

```diff
@@ -254,6 +273,45 @@ func rejectPodAdmissionBasedOnOSField(pod *v1.Pod) bool {
 	return string(pod.Spec.OS.Name) != runtime.GOOS
 }
 
+// rejectPodAdmissionBasedOnSupplementalGroupsPolicy rejects pod only if
+// - the feature is beta or above, and SupplementalPolicy=Strict is set in the pod
+// - but, the node does not support the feature
+//
+// Note: During the feature is alpha or before(not yet released) in emulated version,
+// it should admit for backward compatibility
+func rejectPodAdmissionBasedOnSupplementalGroupsPolicy(pod *v1.Pod, node *v1.Node) bool {
+	admit, reject := false, true // just for readability
+
+	inUse := (pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.SupplementalGroupsPolicy != nil)
+	if !inUse {
+		return admit
+	}
+
+	isBetaOrAbove := false
+	if featureSpec, ok := utilfeature.DefaultMutableFeatureGate.GetAll()[features.SupplementalGroupsPolicy]; ok {
+		isBetaOrAbove = (featureSpec.PreRelease == featuregate.Beta) || (featureSpec.PreRelease == featuregate.GA)
```

Comment:
couldn't this check just be isAlpha? it seems conceptually clearer and we wouldn't need the intermediate variable to make it more readable

Reply:
Sure, I will. Just one thing. Supported emulation verisons in v1.33 binary still include v1.30 that the feature was not yet released. So, I will replace this with like below which does not have intermediate variable only in if statement:

```go
// Admit if this feature is Alpha or Before
if featureSpec, ok := utilfeature.DefaultMutableFeatureGate.GetAll()[features.SupplementalGroupsPolicy]; !ok || (featureSpec.PreRelease == featuregate.Alpha) || (featureSpec.PreRelease == featuregate.PreAlpha) {
	return admit
}
```

Reply:
@haircommander fixed and all the test passed. PTAL 🙇

Reply:
sorry for flip flopping, I think the variable is more readable now that it's such a long condition